azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -226,10 +226,12 @@ func resourceArmKubernetesCluster() *schema.Resource {
 						"client_id": {
 							Type:     schema.TypeString,
 							Required: true,
+							ForceNew: true,
 						},
 
 						"client_secret": {
 							Type:      schema.TypeString,
+							ForceNew:  true,
```

Comment:
This does not have any effect because the secret is not retrieved from the environment nor stored in a state file.